The `fim-files.json` file could not be downloaded during the Wazuh manager package generation

|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 5.0.0-beta4 | Wazuh manager package generation | Manager | - | Debian AMD64 |

The Wazuh manager package for AMD64 Debian systems failed to be created in the 2026-07-09 nightly run due to an error while downloading the `fim-files.json` file from the wazuh-indexer-plugins repository.

- https://github.com/wazuh/wazuh/actions/runs/28986828738/job/86017737071

```
mkdir -p external/indexer-plugins
[-] Downloading dependency fim-files.json ...
curl -so external/indexer-plugins/fim-files.json --cacert /etc/ssl/certs/ca-certificates.crt -f https://raw.githubusercontent.com/wazuh/wazuh-indexer-plugins/refs/heads/5.0.0/plugins/setup/src/main/resources/templates/states/fim-files.json
curl -so external/indexer-plugins/fim-files.json --cacert /etc/ssl/certs/ca-certificates.crt -f https://raw.githubusercontent.com/wazuh/wazuh-indexer-plugins/refs/heads/5.0.0/plugins/setup/src/main/resources/templates/states/fim-files.json
curl -so external/indexer-plugins/fim-files.json --cacert /etc/ssl/certs/ca-certificates.crt -f https://raw.githubusercontent.com/wazuh/wazuh-indexer-plugins/refs/heads/main/plugins/setup/src/main/resources/templates/states/fim-files.json
curl -so external/indexer-plugins/fim-files.json --cacert /etc/ssl/certs/ca-certificates.crt -f https://raw.githubusercontent.com/wazuh/wazuh-indexer-plugins/refs/tags/v5.0.0/plugins/setup/src/main/resources/templates/states/fim-files.json
ERROR: Failed to download fim-files.json from any reference
Makefile:686: recipe for target 'external/indexer-plugins/fim-files.json' failed
```

The file is accessible in the expected version (5.0.0 branch) and can be downloaded successfully. We need to investigate this further to determine whether the `cacert` option is causing the connection to fail.

The same is happening for the Wazuh agent DEB package https://github.com/wazuh/wazuh/issues/37535

A replay generated the package without errors https://github.com/wazuh/wazuh/actions/runs/29004395242, we may need to add some retry task to avoid this problem

--- a/src/Makefile
+++ b/src/Makefile
@@ -401,6 +401,15 @@ ifneq (,$(shell curl --help all 2>/dev/null | grep -F -- '--retry-all-errors'))
 	TZDATA_CURL_FLAGS += --retry-all-errors
 endif
 TZDATA_CURL := curl $(TZDATA_CURL_FLAGS) -o
+# Curl flags for indexer templates and WCS flat files with retry logic
+INDEXER_CURL_FLAGS := -fsS --retry 5 --retry-delay 2 --connect-timeout 20 --max-time 300
+ifneq (,$(shell curl --help all 2>/dev/null | grep -F -- '--retry-connrefused'))
+	INDEXER_CURL_FLAGS += --retry-connrefused
+endif
+ifneq (,$(shell curl --help all 2>/dev/null | grep -F -- '--retry-all-errors'))
+	INDEXER_CURL_FLAGS += --retry-all-errors
+endif
+INDEXER_CURL := curl $(INDEXER_CURL_FLAGS) -o
 DEPS_VERSION = 99-37361
 RESOURCES_URL_BASE := https://packages.wazuh.com/deps/
 RESOURCES_URL := $(RESOURCES_URL_BASE)$(DEPS_VERSION)
@@ -670,8 +679,8 @@ $(INDEXER_TEMPLATES_SERVER_STREAMS:%=$(INDEXER_PLUGINS_DIR)/%): | $(INDEXER_PLUG
 	@success=0; \
 	for ref in $(INDEXER_PLUGIN_REFS_LIST); do \
 		url="$(INDEXER_PLUGINS_BASE_URL)/$$ref/$(INDEXER_PLUGINS_STREAMS_PATH)/$(notdir $@)"; \
-		echo "$(CURL) $@ $(CACERT_OPT) -f $$url"; \
-		if $(CURL) $@ $(CACERT_OPT) -f "$$url" 2>/dev/null; then \
+		echo "$(INDEXER_CURL) $@ $(CACERT_OPT) -f $$url"; \
+		if $(INDEXER_CURL) $@ $(CACERT_OPT) -f "$$url"; then \
 			success=1; \
 			break; \
 		fi; \
@@ -687,8 +696,8 @@ $(INDEXER_PLUGINS_DIR)/%.json: | $(INDEXER_PLUGINS_DIR)
 	 @success=0; \
 	for ref in $(INDEXER_PLUGIN_REFS_LIST); do \
 		url="$(INDEXER_PLUGINS_BASE_URL)/$$ref/$(INDEXER_PLUGINS_STATES_PATH)/$(notdir $@)"; \
-		echo "$(CURL) $@ $(CACERT_OPT) -f $$url"; \
-		if $(CURL) $@ $(CACERT_OPT) -f "$$url" 2>/dev/null; then \
+		echo "$(INDEXER_CURL) $@ $(CACERT_OPT) -f $$url"; \
+		if $(INDEXER_CURL) $@ $(CACERT_OPT) -f "$$url"; then \
 			success=1; \
 			break; \
 		fi; \
@@ -709,8 +718,8 @@ $(WCS_FLAT_DIR)/wcs_flat.yml: | $(WCS_FLAT_DIR)
 	 @success=0; \
 	for ref in $(INDEXER_PLUGIN_REFS_LIST); do \
 		url="$(INDEXER_PLUGINS_BASE_URL)/$$ref/$(WCS_FLAT_STATELESS_PATH)/$(notdir $@)"; \
-		echo "$(CURL) $@ $(CACERT_OPT) -f $$url"; \
-		if $(CURL) $@ $(CACERT_OPT) -f "$$url" 2>/dev/null; then \
+		echo "$(INDEXER_CURL) $@ $(CACERT_OPT) -f $$url"; \
+		if $(INDEXER_CURL) $@ $(CACERT_OPT) -f "$$url"; then \
 			success=1; \
 			break; \
 		fi; \

